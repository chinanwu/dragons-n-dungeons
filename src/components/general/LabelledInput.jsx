import React from 'react';
import PropTypes from 'prop-types';

import Input from './Input.jsx';

import './LabelledInput.less';

export const LabelledInput = ({
  id,
  label,
  type,
  value,
  onChange,
  onKeyDown,
}) => {
  return (
    <div className="LabelledInput">
      <label className="LabelledInput__label" htmlFor={id}>
        {label}
      </label>
      <Input
        id={id}
        className={'LabelledInput__' + type}
        type={type}
        value={value}
        onChange={onChange}
        onKeyDown={onKeyDown}
      />
    </div>
  );
};

LabelledInput.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onChange: PropTypes.func,
  onKeyDown: PropTypes.func,
};

export default LabelledInput;
