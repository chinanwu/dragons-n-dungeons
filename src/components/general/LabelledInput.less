@import '../styling/colours.less';
@import '../styling/fonts.less';

.LabelledInput {
  display: flex;
  flex-direction: column;
  margin-top: 6px;
}

.LabelledInput__label {
  font-size: @font-12;
  color: @input-label-colour;
}

.LabelledInput__text {
  font-size: @font-24;
}
