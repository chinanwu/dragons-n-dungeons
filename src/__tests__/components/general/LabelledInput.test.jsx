import { shallow } from 'enzyme';
import React from 'react';

import { LabelledInput } from '../../../components/general/LabelledInput.jsx';

jest.unmock('../../../components/general/LabelledInput.jsx');

describe('LabelledInput component', () => {
  describe('rendering', () => {
    it('renders LabelledInput', () => {
      const wrapper = shallow(<LabelledInput id="test" />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });
});
