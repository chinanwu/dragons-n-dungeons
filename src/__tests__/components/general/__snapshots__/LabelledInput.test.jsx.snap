// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelledInput component rendering renders LabelledInput 1`] = `
<div
  className="LabelledInput"
>
  <label
    className="LabelledInput__label"
    htmlFor="test"
  />
  <Input
    className="LabelledInput__undefined"
    id="test"
  />
</div>
`;
